src/state/apis/swappers/swappersApi.ts

```diff
@@ -1,9 +1,88 @@
 import { createApi } from '@reduxjs/toolkit/dist/query/react'
+import type { Result } from '@sniptt/monads'
+import { orderBy } from 'lodash'
+import type { GetTradeQuoteInput, SwapErrorRight, TradeQuote2 } from 'lib/swapper/api'
+import { SwapperName } from 'lib/swapper/api'
+import { getInputOutputRatioFromQuote } from 'state/apis/swappers/helpers/getInputOutputRatioFromQuote'
+import { getLifiTradeQuoteHelper } from 'state/apis/swappers/helpers/getLifiTradeQuoteApiHelper'
+import { getThorTradeQuoteHelper } from 'state/apis/swappers/helpers/getThorTradeQuoteApiHelper'
+import type { ApiQuote } from 'state/apis/swappers/types'
+import { isCrossAccountTradeSupported } from 'state/helpers'
+import type { ReduxState } from 'state/reducer'
+import type { FeatureFlags } from 'state/slices/preferencesSlice/preferencesSlice'
+import { selectFeatureFlags } from 'state/slices/selectors'
+import { tradeQuoteSlice } from 'state/slices/tradeQuoteSlice/tradeQuoteSlice'
 
 import { BASE_RTK_CREATE_API_CONFIG } from '../const'
 
 export const swappersApi = createApi({
   ...BASE_RTK_CREATE_API_CONFIG,
   reducerPath: 'swappersApi',
-  endpoints: _build => ({}),
+  endpoints: build => ({
+    getTradeQuote: build.query<ApiQuote[], GetTradeQuoteInput>({
+      queryFn: async (getTradeQuoteInput: GetTradeQuoteInput, { getState, dispatch }) => {
+        const queryStartTime = Date.now()
+        const state = getState() as ReduxState
+        const { sendAddress, receiveAddress } = getTradeQuoteInput
+        const isCrossAccountTrade = sendAddress !== receiveAddress
+        const { LifiSwap, ThorSwap }: FeatureFlags = selectFeatureFlags(state)
+        const quotes: (Result<TradeQuote2, SwapErrorRight> & {
+          swapperName: SwapperName
+        })[] = []
+        const quoteHelperArgs = [getTradeQuoteInput, state] as const
+        if (LifiSwap)
+          try {
+            const lifiTradeQuote: Result<TradeQuote2, SwapErrorRight> =
+              await getLifiTradeQuoteHelper(...quoteHelperArgs)
+            quotes.push({
+              ...lifiTradeQuote,
+              swapperName: SwapperName.LIFI,
+            })
+          } catch (error) {
+            console.error('[getLifiTradeQuoteHelper]', error)
+          }
+        if (ThorSwap)
+          try {
+            const thorTradeQuote: Result<TradeQuote2, SwapErrorRight> =
+              await getThorTradeQuoteHelper(...quoteHelperArgs)
+            quotes.push({
+              ...thorTradeQuote,
+              swapperName: SwapperName.Thorchain,
+            })
+          } catch (error) {
+            console.error('[getThorTradeQuoteHelper]', error)
```

Comment:
there errors should return the error RTK query monad `return { error }`

Reply:
if the intent is to return quotes that succeed thats cool, but uncaught errors should cause the whole request to fail, like a 500 error in an api

Reply:
handled errors are managed with our monads

Reply:
Ah, I forgot to come back to this - something inside thorswapper2 was consistently throwing for some chains (and not via a monad, uncaught throwing).
If `getThorTradeQuoteHelper` throws and it's not wrapped in a `try...catch` the whole RTK request fails.

What we really want to do is return the swapper and the error - proposing I do this in a follow-up PR to maintain velocity